From 5fc681a559422dcc1eb998652b6fc7edcfb693e5 Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Mon, 23 Feb 2015 13:59:05 -0600
Subject: [PATCH 12/13] remoteproc/pruss: set recovery_disabled flag properly

The error recovery feature is not supported for the PRU
remote processors. The flag is a negative flag, so mark
the rproc configuration flag appropriately for PRUs.

The debugfs interface behavior is unchanged, as there
are no true triggers for recovery anyways for the PRU
processors.

Signed-off-by: Suman Anna <s-anna@ti.com>
---
 drivers/remoteproc/pruss_remoteproc.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/remoteproc/pruss_remoteproc.c b/drivers/remoteproc/pruss_remoteproc.c
index 0228620..b6fdb51 100644
--- a/drivers/remoteproc/pruss_remoteproc.c
+++ b/drivers/remoteproc/pruss_remoteproc.c
@@ -830,6 +830,9 @@ static int pru_rproc_probe(struct platform_device *pdev)
 		dev_err(dev, "rproc_alloc failed\n");
 		return -ENOMEM;
 	}
+	/* error recovery is not supported for PRUs */
+	rproc->recovery_disabled = true;
+
 	pru = rproc->priv;
 	pru->id = pdata->id;
 	pru->pruss = platform_get_drvdata(ppdev);
-- 
1.7.9.5

